test/files/pos/t4225.scala

```diff
@@ -0,0 +1,55 @@
+object Test {
+  class Foo {
+    class Bar
+    object Bar {
+      implicit def mkBar: Bar = new Bar
+    }
+  }
+
+  object f extends Foo
+
+  object ImplicitClass {
+    implicit class Ops[F <: Foo](val f0: F) {
+      def op0(implicit b: f0.Bar): f0.Bar = b
+      def op1(i: Int)(implicit b: f0.Bar): f0.Bar = b
+      def op2(i: Int)(j: Boolean)(implicit b: f0.Bar): f0.Bar = b
+      def op3[T](implicit b: f0.Bar): f0.Bar = b
+      def op4(i: => Int)(implicit b: f0.Bar): f0.Bar = b
+    }
+
+    f.op0
+    f.op1(23)
+    f.op2(23)(true)
+    f.op3[Int]
+    f.op4(23)
```

Comment:
Could we introduce a `run` test to show how by-name params interact with this lifting? I'd imagine that we'd need to lift a def or a val with a `Function0` to preserve eval semantics.

Reply:
If the qualifier is passed as a by-name arg to the implicit conversion, it doesn't seem suitably stable for a path (or equivalently, to be liftable to a val)

Reply:
Agreed. That would have to be excluded.